src/Uno.UI/DataBinding/BindingExpression.cs

```diff
@@ -739,7 +739,7 @@ private void SetTargetValueSafe(object v, bool useTargetNullValue)
 		[MethodImpl(MethodImplOptions.AggressiveInlining)]
 		private void InnerSetTargetValueSafe(object v, bool useTargetNullValue)
 		{
-			if (v is UnsetValue)
+			if (v == DependencyProperty.UnsetValue)
```

Comment:
So, `object.ReferenceEquals` is basically `ceq` instruction:

![image](https://github.com/unoplatform/uno/assets/31348972/962cd792-fadd-46f0-a991-f6707d1e8e9e)

(its implementation is just `left == right` (where compile-time type of both is `object`)

For this callsite, this is the IL with the PR, which is just `ceq` (equivalent to ReferenceEquals)

![image](https://github.com/unoplatform/uno/assets/31348972/3cb964bb-f6f8-4ca1-8fdc-f73b5f22b4c9)